src/components/EventCardsGridContainer/index.js

```diff
@@ -77,23 +88,99 @@ class EventCardsGrid extends React.Component {
     } = nextProps;
 
     if (eventStatusIndex !== this.props.eventStatusIndex
-      || sortBy !== this.props.sortBy
-      || syncBlockNum !== this.props.syncBlockNum) {
-      this.executeGraphRequest(eventStatusIndex, sortBy);
+      || sortBy !== this.props.sortBy) {
+      this.executeGraphRequest(eventStatusIndex, sortBy, LIMIT, SKIP);
+      this.setState({ skip: 0 });
+    }
+    if (syncBlockNum !== this.props.syncBlockNum && this.props.syncBlockNum !== 0) {
+      this.setState({ hint: true });
+      setTimeout(() => {
+        this.setState({ hint: false });
+      }, 3000);
     }
   }
 
-  render() {
+  loadMoreOracles = () => {
+    let { skip } = this.state;
     const {
-      theme,
       eventStatusIndex,
-      getTopicsReturn,
-      getOraclesReturn,
+      sortBy,
+      lastUsedAddress,
+      getMoreOracles,
+      getMoreTopics,
     } = this.props;
+    skip += LIMIT;
+
+    this.setState({ skip });
+    const sortDirection = this.props.sortBy || SortBy.Ascending;
 
-    const topics = getTopicsReturn;
-    const oracles = getOraclesReturn;
-    let rowItems;
+    switch (eventStatusIndex) {
+      case EventStatus.Bet: {
+        getMoreOracles(
+          [
+            { token: Token.Qtum, status: OracleStatus.Voting },
+            { token: Token.Qtum, status: OracleStatus.Created },
+          ],
+          { field: 'endTime', direction: sortDirection },
+          LIMIT,
+          skip,
+        );
+        break;
+      }
+      case EventStatus.Set: {
+        getMoreOracles(
+          [
+            { token: Token.Qtum, status: OracleStatus.WaitResult, resultSetterQAddress: lastUsedAddress },
+            { token: Token.Qtum, status: OracleStatus.OpenResultSet },
+          ],
+          { field: 'resultSetEndTime', direction: sortDirection },
+          LIMIT,
+          skip,
+        );
+        break;
+      }
+      case EventStatus.Vote: {
+        getMoreOracles(
+          [
+            { token: Token.Bot, status: OracleStatus.Voting },
+          ],
+          { field: 'endTime', direction: sortDirection },
+          LIMIT,
+          skip,
+        );
+        break;
+      }
+      case EventStatus.Finalize: {
+        getMoreOracles(
+          [
+            { token: Token.Bot, status: OracleStatus.WaitResult },
+          ],
+          { field: 'endTime', direction: sortDirection },
+          LIMIT,
+          skip,
+        );
+        break;
+      }
+      case EventStatus.Withdraw: {
+        getMoreTopics(
+          [
+            { status: OracleStatus.Withdraw },
+          ],
+          { field: 'blockNum', direction: sortDirection },
+          LIMIT,
+          skip,
+        );
+        break;
+      }
+      default: {
+        throw new RangeError(`Invalid tab position ${eventStatusIndex}`);
+      }
+    }
+  }
+
+  render() {
+    const { theme, eventStatusIndex, topics, oracles, sortBy, classes } = this.props; // eslint-disable-line
```

Comment:
put these on separate lines. you should only disable eslint rules if necessary. in this case, there's not downside to having them on separate lines.